webpack/assets/javascripts/react_app/components/TemplateGenerator/TemplateGeneratorActions.js

```diff
@@ -0,0 +1,67 @@
+import { saveAs } from 'file-saver';
+import API from '../../API';
+
+import {
+  TEMPLATE_GENERATE_REQUEST,
+  TEMPLATE_GENERATE_POLLING,
+  TEMPLATE_GENERATE_SUCCESS,
+  TEMPLATE_GENERATE_FAILURE,
+} from './TemplateGeneratorConstants';
+
+const defaultPollingInterval = 2500;
+const pollingInterval =
+  process.env.NOTIFICATIONS_POLLING || defaultPollingInterval;
+
+export const generateTemplate = (url, templateInputData) => dispatch => {
+  dispatch({
+    type: TEMPLATE_GENERATE_REQUEST,
+    payload: { ...templateInputData },
+  });
+  return API.post(url, templateInputData)
+    .then(({ data }) => {
+      dispatch(pollReportData(data.data_url));
+    })
+    .catch(error =>
+      dispatch({
+        type: TEMPLATE_GENERATE_FAILURE,
+        payload: { error, item: templateInputData },
+      })
+    );
+};
+
+const _downloadFile = response => {
+  const blob = new Blob([response.data], {
+    type: response.headers['content-type'],
+  });
+  const filename = response.headers['content-disposition'].match(
+    /filename="(.*)"/
+  );
+  saveAs(blob, (filename && filename[1]) || 'report.txt');
+};
+
+const _getErrors = errorResponse => {
+  if (!errorResponse || !errorResponse.data) return null;
+  if (errorResponse.status === 422) return errorResponse.data.errors;
+  if (errorResponse.data.error) return [errorResponse.data.error]; // most of >500
+  return [errorResponse.data];
+};
+
+export const pollReportData = pollUrl => dispatch => {
```

Comment:
we have other actions for polling, I wonder about creating a general action for polling requests

Reply:
I guess we've got just notifications.
Anyway I agree we should, but I would prefer to do that as a followup, as I don't feel it like a blocker, but if it is for you, I can do that.

Reply:
In subscriptions page we also use the same polling mechanism (which caused some annoying bugs),  it's not a blocker, but I concern that it would be missed.
Having a middlewere for networking is important, especially when we got complex sceanrio like polling.

 @sharvit has refactored the polling logic in katello, but it haven't merged yet, if you have time you can reuse it (https://github.com/sharvit/katello/commit/967fceae2cf2fb9db5870251a17a3a3360710d32#diff-b9290ce729f3b4b0a868fb93e92be9fb)